exercises/luhn/example.go

```diff
@@ -1,27 +1,28 @@
 package luhn
 
-const testVersion = 1
+import "strings"
 
-func Valid(n string) bool {
-	if len(n) == 1 {
-		return false
-	}
-	d := extract(n)
-	if len(d) == 0 {
+const testVersion = 2
+
+func Valid(id string) bool {
+
+	if len(strings.TrimSpace(id)) == 1 {
 		return false
 	}
-	last := len(d) - 1
-	return (check(d[:last])+d[last])%10 == 0
-}
 
-func extract(n string) []int {
-	d := make([]int, 0, len(n))
-	for _, r := range n {
+	d := make([]int, 0, len(id))
+
+	for _, r := range id {
+		if r != ' ' && (r < '0' || r > '9') {
+			return false
+		}
 		if r >= '0' && r <= '9' {
 			d = append(d, int(r-'0'))
 		}
 	}
-	return d
+
+	last := len(d) - 1
+	return (check(d[:last])+d[last])%10 == 0
```

Comment:
I'm assuming `check` returns the sum? Would anything go wrong if we just used `check(d) % 10 == 0`?

E: Oh never mind, this is a same line as above, just moved. I guess we don't need to change it, though if it's going to show up in git diff as a diff anyway, maybe we could consider